osquery/database/database.cpp

```diff
@@ -343,4 +350,51 @@ void dumpDatabase() {
     }
   }
 }
+
+Status updateDatabase() {
+  std::vector<std::string> keys;
+  auto s = scanDatabaseKeys(kQueries, keys);
+  if (!s.ok()) {
+    return Status(1, "Failed to lookup legacy query data from database");
+  }
+
+  for (const auto& key : keys) {
+    std::string value;
+    if (!getDatabaseValue(kQueries, key, value)) {
+      LOG(WARNING) << "Failed to get value from database " << key << ": "
+                   << value;
+      continue;
+    }
+
+    pt::ptree tree;
+    try {
+      std::stringstream ss;
+      ss << value;
+      pt::read_json(ss, tree);
+    } catch (const pt::json_parser::json_parser_error& /* e */) {
+      LOG(WARNING) << "Conversion from ptree to RapidJSON failed for " << key
```

Comment:
This might be better as a `LOG(INFO)`, as this will be triggered for values in the DB that _do not_ require updating. Either that or this could just be a blank catch statement?

Reply:
What is the behaviour of tree is parsing fails? I think maybe there should be an explicit continue here.

Reply:
@obelisk yeah i think you're right, I think I forgot that `continue`